### PR TITLE
SUP-1788: CircleCI docs tabulation

### PR DESCRIPTION
**Pull request by james2791**
Transformed the CircleCI docs into a markdown tabulated format, with each main `key` of a Circle pipeline and all sub-key support listed (with touched up notes from the original)

### CODE REVIEW

File: docs/CircleCI.md [hunk 1]
```diff
@@ -1,72 +1,82 @@
...
-Additionally, an approach to assist in configuring reusable pieces of configuration within a Buildkite pipeline is through the use of [YAML anchors](https://buildkite.com/docs/plugins/using#using-yaml-anchors-with-plugins) - which are defined and used in a similar way to CircleCI `aliases`. 
\ No newline at end of file
+The Buildkite Migration tool's currently supported (🟢), partially supported (🟡) and unsupported (🔴) properties in translation of CircleCI pipelines to Buildkite pipelines are listed below.
+
+### Logical Operators/Helper Keys
+
+| Key | Supported? | Notes |
+| --- | --- | --- |
+| `aliases` | 🟢 | A list of reusable YAML snippets of a CircleCI pipeline.
+| `aliases.&<name>` | 🟢 | A singular alias describing a resusable snippet of YAML to apply to a specific point in a CircleCI pipeline. Defined with a `&` (anchor) - these blocks are substituted into configuration with `*`: for example, `*tests`.
```
Comment (toote) [inline docs/CircleCI.md:10]: I'm not sure these two rows should be in the document. In this section for sure not as they are not actual keys nor CircleCI-specific, just part of the YAML standard.

Maybe just add a note that anchors and aliases will be respected and parsed correctly.

**Review by toote — CHANGES_REQUESTED** [review]
Some other changes to the content and support level itself.

File: docs/CircleCI.md [hunk 2]
```diff
@@ -1,72 +1,82 @@
...
+| `jobs.<name>.environment` | 🟢 | The `job` level environment variables of a CircleCI pipeline. Applied in the generated [command step](https://buildkite.com/docs/pipelines/command-step) as [step level](https://buildkite.com/docs/pipelines/environment-variables#runtime-variable-interpolation) environment variables with the `env` key. |
+| `jobs.<name>.macos` | 🟢 | Specifies that the `job` will run on a macOS bases execution environment. The [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) tags for the generated [command step](https://buildkite.com/docs/pipelines/command-step) will be set to `executor_type: osx`, as well as the specified version of `xcode` from the `macos` parameters as `executor_xcode: <version>`. |
```
[the remark this answers is not in the document]
Comment (james2791) [reply docs/CircleCI.md]: I've split these off into their own table under a H4 `Executors` table

File: docs/CircleCI.md [hunk 3]
```diff
@@ -1,72 +1,82 @@
...
+| `jobs.<name>.shell` | 🔴| The `shell` property sets the default shell that is used across all commands within all steps. This should be configured on the agent - or by defining the `shell` [option](https://buildkite.com/docs/agent/v3/cli-start#shell) when starting a Buildkite agent: which will set the shell command used to interpret all build commands. |
+| `jobs.<name>.steps` | 🟢 | A collection of commands that are executed as part of a CircleCI `job`. Steps can be defined within an `alias`. All `steps` within a singular `job` are translated to the `commands` of a shared [command step](https://buildkite.com/docs/pipelines/command-step) within the generated Buildkite pipeline to ensure they share the same execution environment. |
+| `jobs.<name>.windows` | 🟢 | Specifies that the `job` will run on a Windows based execution environment. The [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) tags for the generated [command step](https://buildkite.com/docs/pipelines/command-step) will be set to `executor_type: windows`. |
```
[the remark this answers is not in the document]
Comment (james2791) [reply docs/CircleCI.md]: I've split these off into their own table under a H4 `Executors` table

File: docs/CircleCI.md [hunk 4]
```diff
@@ -1,72 +1,82 @@
...
+| `jobs.<name>.parameters` | 🟢 | Reusable keys that are used within `step` definitions within a `job`. Default parameters that are specified in a `parameters` block are passed through into the [command step](https://buildkite.com/docs/pipelines/command-step)'s `commands` if specified. |
+| `jobs.<name>.resource_class` | 🟢 | The specification of compute that the executor will require in running a job. This is used to specify the `resource_class` [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) tag for the corresponding [command step](https://buildkite.com/docs/pipelines/command-step). |
```
[the remark this answers is not in the document]
Comment (james2791) [reply docs/CircleCI.md]: I've split these off into their own table under a H4 `Executors` table

File: docs/CircleCI.md [hunk 5]
```diff
@@ -1,72 +1,82 @@
...
+
+| Key | Supported? | Notes |
+| --- | --- | --- |
+| `commands` | 🟢 | A `command` defined in a CircleCI pipeline is a reusable set of instructions with parameters that can be inserted into required `job` executions. Commands can have their own set list of `steps` that are translated through to the generated [command step](https://buildkite.com/docs/pipelines/command-step)'s `commands`. If a `command` contains a `parameters` key, they are respected when used in jobs/workflows and their defaults values used when not specified. |
+
+### Executors (`executors`)
+
+| Key | Supported? | Notes |
+| --- | --- | --- |
+| `executors` | 🔴 | The `executor` key is currently not supported. The execution environment in Buildkite is defined by the [queues](https://buildkite.com/docs/agent/v3/queues#setting-an-agents-queue) and [tags](https://buildkite.com/docs/agent/v3/cli-start#setting-tags) applied to an agent, and [targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) them when creating builds. |
```
[the remark this answers is not in the document]
Comment (james2791) [reply docs/CircleCI.md]: Tweaked. thanks @toote 👍

File: docs/CircleCI.md (hunk 1 above)
Comment (james2791) [reply docs/CircleCI.md]: Agreed - I've split aliases/anchors which now lives in a note 👍

File: docs/CircleCI.md [hunk 6]
```diff
@@ -1,72 +1,82 @@
...
+| --- | --- | --- |
+| `version` | 🔴 | The version of the CircleCI pipeline configuration applied to this pipeline. No equivalent mapping exists in Buildkite. Attributes for required and optional attributes in the various step types supported in Buildkite are listed in the [docs](https://buildkite.com/docs/pipelines/step-reference). |
+
+### Workflows (`workflows`)
+
+| Key | Supported? | Notes |
+| --- | --- | --- |
+| `workflows` | 🟢 | A a collection of `jobs`, whose ordering defines how a CircleCI pipeline is run. |
+| `workflows.<name>` | 🟢 | An induvidual named workflow that makes up part of the CircleCI pipeline's definition. If a CircleCI pipeline has more than 1 `workflow` specified, each will be transitioned to a [group step](https://buildkite.com/docs/pipelines/group-step). |
+| `workflows.<name>.jobs` | 🟢 | Induvidual named `jobs` that make up part of this specific workflow. Each `job` that is defined as part of a `workflow` within a CircleCI pipeline will be transitioned to a Buildkite [command step](https://buildkite.com/docs/pipelines/command-step) within the generated pipeline. |
```
[the remark this answers is not in the document]
Comment (james2791) [reply docs/CircleCI.md]: I called this out deliberately in the newest commit where workflow jobs from orbs, and using orbs in a specific `job.<name>.steps` isn't supported (but approvals for the former are etc)

**Comment by james2791** [comment]
@toote as with #79, I've tweaked the docs to use a combination of ✅, ⚠️ and ❌ to compensate for the coloured circles for now - let me know what you think!

File: docs/CircleCI.md [hunk 7]
```diff
@@ -1,72 +1,90 @@
...
+| `jobs.<name>.shell` | ❌ | The `shell` property sets the default shell that is used across all commands within all steps. This should be configured on the agent - or by defining the `shell` [option](https://buildkite.com/docs/agent/v3/cli-start#shell) when starting a Buildkite agent: which will set the shell command used to interpret all build commands. |
+| `jobs.<name>.steps` | ✅ | A collection of non-`orb` `jobs` commands that are executed as part of a CircleCI `job`. Steps can be defined within an `alias`. All `steps` within a singular `job` are translated to the `commands` of a shared [command step](https://buildkite.com/docs/pipelines/command-step) within the generated Buildkite pipeline to ensure they share the same execution environment. |
+| `jobs.<name>.working_directory` | ✅ | The location of the executor where steps are run in. If set, a change directory (`cd`) command is created within the shared `commands` of a Buildkite [command step](https://buildkite.com/docs/pipelines/command-step) to the desired location. |
+
+#### Executors 
+
+| Key | Supported? | Notes |
```
Comment (toote) [inline docs/CircleCI.md:45]: I would add a note here stating that, while the translation sets up the correct targeting, for them to work they need to setup their agents with the necessary dependencies and tools (maybe using the AWS elastic stack).

**Review by toote — APPROVED** [review]
Just a minor suggestion wrt executors, but awesome work!

File: docs/CircleCI.md (hunk 7 above)
Comment (james2791) [reply docs/CircleCI.md]: Just added in 👍

**Pull request merged**